libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m

```diff
@@ -776,9 +776,11 @@ - (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigati
 }
 
 - (BOOL)shouldUpdateDomain:(NSURL *)webviewURL {
-    NSURL *currentDomain = [NSURL URLWithString:self.credentials.domain];
     NSString *myDomainSuffix = @".my.salesforce.com";
-    return (![currentDomain.host hasSuffix:myDomainSuffix] && [webviewURL.host hasSuffix:myDomainSuffix]);
+    return (![self.credentials.domain hasSuffix:myDomainSuffix] &&
+            [webviewURL.host hasSuffix:myDomainSuffix] &&
+            [webviewURL.path isEqualToString:@"/"] &&
```

Comment:
@brandonpage we should check for an empty path on Android as well, as it is done here (line 782).